python/lsst/dax/distribution/testingdistrib.py

```diff
@@ -100,35 +111,41 @@ class ClientTestThrd(threading.Thread):
     """Class for testing the client side of messaging
     """
 
-    def __init__(self, host, port, name, arg_string, cfg_file_contents,
-                 maxCount, chunkListA, pCfgFiles, ingest_dict):
+    def __init__(self, host, port, name, objects, visits, seed, cfg_file_contents,
+                 maxCount, chunkListA, pCfgFiles, ingest_dict, timing_dict):
         super().__init__()
         self.success = None
         self.warnings = 0
         self.host = host
         self.port = port
         self.name = name
-        self.arg_string = arg_string
+        self.objects = objects
+        self.visits = visits
+        self.seed = seed
         self.cfg_file_contents = cfg_file_contents
         self.maxCount = maxCount
         self.chunkListA = chunkListA
         self.pCfgFiles = pCfgFiles
         self.ingest_dict = ingest_dict
+        self.timing_dict = timing_dict
 
     def run(self):
         with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
             s.connect((self.host, self.port))
             client = DataGenConnection.DataGenConnection(s)
             client.clientReqInit()
-            name, arg_string, cfg_file_contents, ingest_dict = client.clientRespInit()
+            name, objects, visits, seed, cfg_file_contents, ingest_dict = client.clientRespInit()
             print("ingest_dict=", ingest_dict)
-            if (name == self.name and arg_string == self.arg_string
+            if (name == self.name
+                and objects == self.objects
+                and visits == self.visits
+                and seed == self.seed
```

Comment:
An explanation of the long and statement here would help.

Reply:
I added a comment. If the values sent from the server thread across the connection don't match what is expected, there's a problem with encoding/decoding the messages.